src/test/librbd/test_librbd.cc

```diff
@@ -1941,7 +1951,9 @@ TEST_F(TestLibRBD, TestIO)
   rados_ioctx_create(_cluster, m_pool_name.c_str(), &ioctx);
 
   bool skip_discard = is_skip_partial_discard_enabled();
-
+  if (is_feature_enabled(RBD_FEATURE_EXCLUSIVE_LOCK)) {
```

Comment:
Nit: why doesn't RWL respect skipping partial discard?

Reply:
This is related to discard granularity size. When using RWL, the prune is done in image level, we have no idea about the object. Please look at ImageDiscardRequest<I>::prune_object_extents().  
So I set m_image_ctx.discard_granularity_bytes = 0 if rwl is enabled.